dbms/src/Flash/Mpp/BroadcastOrPassThroughWriter.cpp

```diff
@@ -69,18 +69,18 @@ template <class StreamWriterPtr>
 template <bool send_exec_summary_at_last>
 void BroadcastOrPassThroughWriter<StreamWriterPtr>::encodeThenWriteBlocks()
 {
-    TrackedMppDataPacket tracked_packet(current_memory_tracker);
+    auto tracked_packet = std::make_shared<TrackedMppDataPacket>();
     if constexpr (send_exec_summary_at_last)
     {
         TrackedSelectResp response;
         summary_collector.addExecuteSummaries(response.getResponse(), /*delta_mode=*/false);
-        tracked_packet.serializeByResponse(response.getResponse());
+        tracked_packet->serializeByResponse(response.getResponse());
     }
     if (blocks.empty())
     {
         if constexpr (send_exec_summary_at_last)
         {
-            writer->write(tracked_packet.getPacket());
+            writer->write(tracked_packet);
```

Comment:
Using `TrackedMppDataPacket`, we can avoid `mpp::MPPDataPacket` being copied multiple times in the mpp tunnel.